src/main/java/edu/agh/wfiis/designpatterns/patternsUsageExample/design_patterns/DesignPatternFactory.java

```diff
@@ -0,0 +1,35 @@
+package edu.agh.wfiis.designpatterns.patternsUsageExample.design_patterns;
+
+
+import java.util.HashMap;
+import java.util.Map;
+
+public class DesignPatternFactory {
+    public static class DesignPatternNotRegisteredException extends Exception {
+        DesignPatternNotRegisteredException(String s) {
+            super(s);
+        }
+    }
+
+    private static Map<String, DesignPattern> registeredDesignPatterns = new HashMap<>();
+
+    static {
+        registerDesignPattern(new BuilderDesignPattern());
+        registerDesignPattern(new DecoratorDesignPattern());
+        registerDesignPattern(new StrategyDesignPattern());
+    }
+
+    private static void registerDesignPattern(DesignPattern designPattern) {
+        registeredDesignPatterns.put(designPattern.getName(), designPattern);
+    }
+
+    public static DesignPattern create(String patternName) throws DesignPatternNotRegisteredException {
```

Comment:
This might be just my lack of experience with this pattern but isn't the name 'create' misleading when it actually never creates objects in this method, but returns always the same reused object. I know this effectively doesn't matter since DesignPattern classes are immutable, but still :P